### PR TITLE
feat(core): reduce the coupling between render and driver

### PR DESCRIPTION
**Pull request by churchill-zhang**
# Pre-PR Checklist

- [x] I added/updated relevant documentation.
- [x] I followed the [Convention Commit](https://conventionalcommits.org/) guideline with maximum 72 characters to submit commit message.
- [x] I squashed the repeated code commits.
- [x] I signed the [CLA].
- [x] I added/updated test cases to check the change I am making.
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by ilikethese — APPROVED** [review]

File: framework/android/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+# region footstone
+target_include_directories(${PROJECT_NAME} PRIVATE $<TARGET_PROPERTY:footstone,INTERFACE_INCLUDE_DIRECTORIES>)
+# endregion
+
+# region jni
+target_include_directories(${PROJECT_NAME} PRIVATE $<TARGET_PROPERTY:jni,INTERFACE_INCLUDE_DIRECTORIES>)
+# endregion
+
+# region utils
+target_include_directories(${PROJECT_NAME} PUBLIC $<TARGET_PROPERTY:utils,INTERFACE_INCLUDE_DIRECTORIES>)
```
Comment (churchill-zhang) [inline framework/android/CMakeLists.txt:49]: 这里不应该是public吧

File: framework/android/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+# region source set
+set(SOURCE_SET_STANDALONE src/framework_native_jni.cc)
+# The linker treats `.a` and `.o` files differently:
+#   * `.o` files from `.a` file only referenced `.o` object files are linked.
+#   * `.o` files in the linker always go inside, referenced or not.
+#
+# The `SOURCE_SET_STANDALONE` including static initializers,
+# so its visibility must be PUBLIC.
+target_sources(${PROJECT_NAME} PUBLIC ${SOURCE_SET_STANDALONE})
```
Comment (churchill-zhang) [inline framework/android/CMakeLists.txt:64]: 理论不需要对外暴露

File: framework/android/src/framework_native_jni.cc [hunk 3]
```diff
@@ -134,14 +134,14 @@ void UpdateRootSize(JNIEnv *j_env, jobject j_object, jint j_instance_id, jint j_
   dom_manager->PostTask(Scene(std::move(ops)));
 }
 
-void UpdateNodeSize(JNIEnv *j_env, jobject j_object, jint j_instance_id,  jint j_root_id, jint j_node_id,
+void UpdateNodeSize(JNIEnv *j_env, jobject j_object, jint j_render_manager_id,  jint j_root_id, jint j_node_id,
                     jfloat j_width, jfloat j_height, jboolean j_is_sync) {
   auto& map = NativeRenderManager::PersistentMap();
```
Comment (churchill-zhang) [inline framework/android/src/framework_native_jni.cc:139]: node size 是 dom的内容，这里需要render干啥

File: framework/android/src/framework_native_jni.cc [hunk 4]
```diff
@@ -180,14 +180,14 @@ void UpdateNodeSize(JNIEnv *j_env, jobject j_object, jint j_instance_id,  jint j
 }
 
 void DoCallBack(JNIEnv *j_env, jobject j_object,
-                jint j_instance_id, jint j_result, jstring j_func_name, jint j_root_id, jint j_node_id,
+                jint j_render_manager_id, jint j_result, jstring j_func_name, jint j_root_id, jint j_node_id,
                 jlong j_cb_id, jbyteArray j_buffer, jint j_offset, jint j_length) {
   auto& map = NativeRenderManager::PersistentMap();
   std::shared_ptr<NativeRenderManager> render_manager;
-  bool ret = map.Find(static_cast<uint32_t>(j_instance_id), render_manager);
+  bool ret = map.Find(static_cast<uint32_t>(j_render_manager_id), render_manager);
```
Comment (churchill-zhang) [inline framework/android/src/framework_native_jni.cc:187]: 这里也没看出来需要renderManager干啥，最终node是从dom_manager取出来的

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/Hippy?pullRequest=2235) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/Hippy?pullRequest=2235) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: churchill-zhang<br/>:x: 谌立<br/><hr/>**谌立** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/Hippy?pullRequest=2235) it.</sub>

**Pull request closed without merging**